invoiceregistry_test.go

```diff
@@ -496,11 +496,93 @@ func TestOverpayment(t *testing.T) {
 	cancelAcceptUpdates()
 }
 
+func TestMPPValidity(t *testing.T) {
+	defer test.Timeout()()
+	t.Parallel()
+
+	c := newRegistryTestContext(t, optAutoSettle(true))
+
+	resolved := make(chan HtlcResolution)
+
+	notifyHtlc := func(id uint64, rHash lntypes.Hash,
```

Comment:
This one is repeated a lot. Maybe for the next time, we can see if it can be extracted to a method on the context. Would save quite a few lines in this file.

Reply:
I've factored out this inner function into a method on the context, set convenient defaults, and added optional arguments for use in tests similar to what we did with starting a registry context. I've also changed all of the tests to now use this method. It added a bit of setup but now there's far less boilerplate in each test.